### ISSUE TITLE
Windows 11 protocol handler does not start Docker-Desktop since 4.27.0 (sign in fails)

### ISSUE THREAD

**Issue by jruhe-adesso** [docker/for-win#13895]
### Description

As a consequence, sign in fails: I cannot sign in anymore.

Trying to start Docker-Desktop from Powershell shows no offect either:

`Start-Process "docker-desktop://auth0/complete-login?code=P71hceZYpcPCp-fLVU8ua3MUUeGQEkrrbF037i6teX-kF&state=ZcCFLfPXXXIFuPlu4zdDr0YtUh3YxutIuz8aXutdqXM"`

### Reproduce

Push sign in button.

### Expected behavior

Auth information is passed from browser to docker-desktop app.

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.10
 Version:           25.0.1
 API version:       1.44
 Go version:        go1.21.6
 Git commit:        29cf629
 Built:             Tue Jan 23 23:08:30 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Desktop
 Engine:
  Version:          25.0.1
  API version:      1.44 (minimum version 1.24)
  Go version:       go1.21.6
  Git commit:       71fa3ab
  Built:            Tue Jan 23 23:09:46 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.27
  GitCommit:        a1496014c916f9e62104b33d1bb5bd03b0858e59
 runc:
  Version:          1.1.11
  GitCommit:        v1.1.11-0-g4bccb38
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    25.0.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.1-desktop.4
    Path:     /usr/local/lib/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.24.3-desktop.1
    Path:     /usr/local/lib/docker/cli-plugins/docker-compose
  debug: Get a shell into any image or container. (Docker Inc.)
    Version:  0.0.22
    Path:     /usr/local/lib/docker/cli-plugins/docker-debug
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.21
    Path:     /usr/local/lib/docker/cli-plugins/docker-extension
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.4
    Path:     /usr/local/lib/docker/cli-plugins/docker-feedback
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.0.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-sbom
  scout: Docker Scout (Docker Inc.)
    Version:  v1.3.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-scout
WARNING: Plugin "/usr/local/lib/docker/cli-plugins/docker-scan" is not valid: failed to fetch metadata: fork/exec /usr/local/lib/docker/cli-plugins/docker-scan: no such file or directory

Server:
 Containers: 15
  Running: 0
  Paused: 0
  Stopped: 15
 Images: 36
 Server Version: 25.0.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: active
  NodeID: q290uer8k3f4okm29kq67hrks
  Is Manager: true
  ClusterID: 9n9hdz8f4f108t822cnevy9lg
  Managers: 1
  Nodes: 1
  Default Address Pool: 10.0.0.0/8
  SubnetSize: 24
  Data Path Port: 4789
  Orchestration:
   Task History Retention Limit: 5
  Raft:
   Snapshot Interval: 10000
   Number of Old Snapshots to Retain: 0
   Heartbeat Tick: 1
   Election Tick: 10
  Dispatcher:
   Heartbeat Period: 5 seconds
  CA Configuration:
   Expiry Duration: 3 months
   Force Rotate: 0
  Autolock Managers: false
  Root Rotation In Progress: false
  Node Address: 192.168.65.3
  Manager Addresses:
   192.168.65.3:2377
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: a1496014c916f9e62104b33d1bb5bd03b0858e59
 runc version: v1.1.11-0-g4bccb38
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 5.15.146.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 20
 Total Memory: 19.53GiB
 Name: docker-desktop
 ID: d4787b34-1c60-4458-94df-ff3fa9bb2645
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

E10849F9-2E98-46A8-A8D7-868E9C31B0FA/20240131131303

**Comment by jtabox** [comment]
Still an issue in v4.27.1, after a complete wipe (everything, incl. Local/Roaming AppData and .docker directory in user folder) and reinstall from scratch.
The menu itself shows me as logged in, but everywhere else it says I'm not signed in. Trying to complete the process won't return me from Edge to Docker, and the error is persisting.

![image](https://github.com/docker/for-win/assets/56894914/e4225566-32f1-49a6-a475-0582729a0114)

**Comment by jruhe-adesso** [comment]
You are right. Taskbar menu considers me as logged in, too.

**Comment by karman-docker** [comment]
@jtabox could you share a diagnostic bundle from your system please. thank you.

**Comment by jtabox** [comment]
I'm not sure what the diagnostic bundle entails. I'm sharing the outputs of `docker info` and `docker version` below, if there's anything else I need to share let me know.

Here's `docker info`:
```
Client:
 Version:    25.0.2
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.1-desktop.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.24.3-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  debug: Get a shell into any image or container. (Docker Inc.)
    Version:  0.0.22
    Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.21
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.0.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.3.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 0
 Server Version: 25.0.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
 Kernel Version: 5.15.133.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 24
 Total Memory: 23.47GiB
 Name: docker-desktop
 ID: 60589ddf-a5d4-4623-a9c7-71aa4372a686
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
```

And here's `docker version`:
```
Client:
 Cloud integration: v1.0.35+desktop.10
 Version:           25.0.2
 API version:       1.44
 Go version:        go1.21.6
 Git commit:        29cf629
 Built:             Thu Feb  1 00:24:09 2024
 OS/Arch:           windows/amd64
 Context:           default

Server: Docker Desktop 4.27.1 (136059)
 Engine:
  Version:          25.0.2
  API version:      1.44 (minimum version 1.24)
  Go version:       go1.21.6
  Git commit:       fce6e0c
  Built:            Thu Feb  1 00:23:17 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.28
  GitCommit:        ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by karman-docker** [comment]
@jtabox  The diagnostics bundle collects the logs and configs needed for troubleshooting. PII is redacted. If you are ok to share it, use the below command.

If the issue is still happening on your system, could you re-attempt clicking sign-in and after that run the below command. This will generate a diagnostic bundle.
`"C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" gather -upload`
Once this command completes it will report the diagnostic ID on the terminal. Please share that ID here. Thank you.

**Comment by karman-docker** [comment]
@jruhe-adesso could you confirm you have this registry key on your system please.
<img width="1066" alt="Screenshot 2024-02-06 at 07 51 30" src="https://github.com/docker/for-win/assets/110832017/5018e954-f7e1-45a7-b539-3a27b5a3138c">

**Comment by jtabox** [comment]
@karman-docker It is indeed still happening, here's the diagnostic bundle info output:

```
Uploaded diagnostics to: https://docker-pinata-support.s3.amazonaws.com/incoming/3/25CEB2BE-D79A-4F2B-ACB7-B6FAC9D4F447/20240206105236.zip
Diagnostics ID:      (uploaded)
```

**Comment by karman-docker** [comment]
Thanks @jtabox for sharing the diagnostics bundle.
From the logs I see that login was successful. Log was attempted around 10:51:16

```
[2024-02-06T10:51:16.404815800Z][com.docker.backend.exe.registry][I] open browser for login

[2024-02-06T10:51:16.405818500Z][com.docker.backend.exe.browser][I] opening url in browser: https://login.docker.com/authorize?audience=https%3A%2F%2Fhub.docker.com&client_id=

[2024-02-06T10:51:16.542021500Z][com.docker.backend.exe.registry][I] login session initialized
[2024-02-06T10:51:24.210052800Z][com.docker.backend.exe.registry][I] login request received with auth0 callback url: docker-desktop://auth0/complete-login?code=
[2024-02-06T10:51:24.210052800Z][com.docker.backend.exe.ipc][I] (ee944bb2) 94dffabd-BackendAPI S->C 738b3115-URLSchemeHandlerCMD POST /registry/complete-login (507.5µs): OK
```

Does the dashboard (UI) still indicate that you are not logged-in ? In that case it may be an issue only with the Dashboard not reflecting the correct state.

Are you able to perform activities that require you to be signed-in and are completing successfully ?

**Comment by karman-docker** [comment]
There is a similar issue reported on Mac https://github.com/docker/for-mac/issues/7160. Closing and reopening the Dashboard correctly reflects the sign-in status. Could you try it @jtabox ? Thank you.

**Comment by wanjuntham** [comment]
> Closing and reopening the Dashboard correctly reflects the sign-in status.

I had the same issue, can prove this is a valid workaround.

**Comment by jtabox** [comment]
@karman-docker Yes, the UI still indicated I wasn't logged in, exactly like the screenshot I posted above. And indeed, after doing what you asked (closing and reopening the Dashboard) the UI now shows that I'm logged in. I must admit it's kind of embarassing I didn't think to do this very simple thing 😅 

I'm having another UI issue that *feels* related, which is that Docker Desktop doesn't seem to respect the option below, i.e. to not open the Dashboard every time it starts. I have it set to start automatically when I log in, so the Dashboard pops up every time I log.

![image](https://github.com/docker/for-win/assets/56894914/7ddeb5ec-69b3-43d4-aac7-dba053ce852a)


-------------
------------
Another question I had:

Looking through those diagnostics logs, I noticed this in `self-diagnose.json`:

```
[FAIL] DD0029: is the WSL 2 Linux filesystem corrupt? [    2.748394] EXT4-fs error (device sdc): ext4_put_super:1196: comm Xwayland: Couldn't clean up the journal
```

Checking `dmesg` in WSL I could find those lines:

```
[    2.741056] blk_update_request: I/O error, dev sdc, sector 0 op 0x1:(WRITE) flags 0x800 phys_seg 0 prio class 0
[    2.746483] Buffer I/O error on dev sdc, logical block 134184960, lost sync page write
[    2.746920] JBD2: Error -5 detected when updating journal superblock for sdc-8.
[    2.747316] Aborting journal on device sdc-8.
[    2.747579] Buffer I/O error on dev sdc, logical block 134184960, lost sync page write
[    2.747971] JBD2: Error -5 detected when updating journal superblock for sdc-8.
[    2.748394] EXT4-fs error (device sdc): ext4_put_super:1196: comm Xwayland: Couldn't clean up the journal
[    2.748846] EXT4-fs (sdc): Remounting filesystem read-only
```

Are those related? Or is it something that I need to somehow fix?

**Comment by karman-docker** [comment]
Hi @jtabox 

> I'm having another UI issue that feels related, which is that Docker Desktop doesn't seem to respect the option below,

That is being tracked in https://github.com/docker/for-mac/issues/7160

For the errors you found in self-diagnose, I suggest quit Docker Desktop and run `wsl --shutdown` command and then restart Docker Desktop, that should handle it.

**Comment by karman-docker** [comment]
@jruhe-adesso can you confirm the workaround of closing and reopening the dashboard (not the app, just the dashboard/UI) resolves the issue for you too ?

**Comment by jtabox** [comment]
@karman-docker They're still there after shutting down Docker Desktop and WSL and then restarting Docker Desktop.

```
[    2.104095] blk_update_request: I/O error, dev sdc, sector 0 op 0x1:(WRITE) flags 0x800 phys_seg 0 prio class 0
[    2.105796] blk_update_request: I/O error, dev sdc, sector 0 op 0x1:(WRITE) flags 0x800 phys_seg 0 prio class 0
[    2.106669] Buffer I/O error on dev sdc, logical block 134184960, lost sync page write
[    2.107254] JBD2: Error -5 detected when updating journal superblock for sdc-8.
[    2.107732] Aborting journal on device sdc-8.
[    2.108038] Buffer I/O error on dev sdc, logical block 134184960, lost sync page write
[    2.108577] JBD2: Error -5 detected when updating journal superblock for sdc-8.
[    2.109124] EXT4-fs error (device sdc): ext4_put_super:1196: comm Xwayland: Couldn't clean up the journal
[    2.109600] EXT4-fs (sdc): Remounting filesystem read-only
```

They also show up right after a restart. `sdc` is Docker's `/wsl/docker-desktop/docker-desktop-user-distro`.

**Comment by karman-docker** [comment]
@jtabox  I suggest track that separately. May be you can add a comment here: https://github.com/docker/for-win/issues/13340

**Comment by jtabox** [comment]
@karman-docker Will do that, thanks for your help here.

**Comment by jruhe-adesso** [comment]
> @jruhe-adesso could you confirm you have this registry key on your system please.
Yes. I can.
![image](https://github.com/docker/for-win/assets/15670023/e1ec86f2-5b05-4de7-8078-e8767bae7533)

**Comment by karman-docker** [comment]
@jruhe-adesso we think this is the same issue as https://github.com/docker/for-mac/issues/7160.
Does closing and reopening the Dashboard reflect the correct sign-in status for you ?

**Comment by jruhe-adesso** [comment]
> @jruhe-adesso can you confirm the workaround of closing and reopening the dashboard (not the app, just the dashboard/UI) resolves the issue for you too ?

Yes. I can. Thank you.

**Comment by davewasthere** [comment]
Experiencing exactly the same issue, and closing and reopening the dashboard UI shows that I am indeed logged in.
So the docker-desktop: protocol handler is working, but just not showing anything when you click on it? 

After this, I did try logging out, then logging back in again, but this time it updated the profile icon to show me I was logged in. And now I can't reproduce the original unresponsiveness where it just continued to show the login button.

**Comment by MihaelaStoica** [comment]
In Docker Desktop 4.27.2 we fixed an issue with the dashboard being shown at startup (https://github.com/docker/for-win/issues/13887); this bug also caused the dashboard to appear as not signed in and the 'sign in' button was not redirecting back to the dashboard correctly.

The issue reported here seem to be the same, so closing this ticket as fixed in 4.27.2.
